components/common/src/main/java/com/hotels/styx/config/schema/Schema.java

```diff
@@ -139,7 +139,7 @@ private static String message(List<String> parents, String self, JsonNode value)
     }
 
     private static void assertNoUnknownFields(String prefix, Schema schema, List<String> fieldsPresent) {
-        Set<String> knownFields = ImmutableSet.copyOf(schema.fieldNames());
+        Set<String> knownFields = copyToUnmodifiableSet(schema.fieldNames());
```

Comment:
`copyToUnmodifiableSet` is quite a long name for a really basic frequent operation. Can we just call it `setOf`? Or something equally succinct?

Reply:
Originally it seemed important to call out that this is making a copy (not using a wrapper), and that it is unmodifiable. But there are other requirements that emerged too - it preserves the original order, and it doesn't allow nulls. So, as these can't really be apparent in the name as well, it's probably less important that the copy and immutability are also apparent. So yeah - `setOf` works for me.

Reply:
Obviously the same for `listOf`?

Reply:
> Originally it seemed important to call out that this is making a copy (not using a wrapper), and that it is unmodifiable.

I would document this in the API contract instead of the method name. I think taking a defensive copy and being unmodifiable are sensible defaults. IMHO it should be called out in the method name when it merely wraps or keeps the returned list modifiable.

Reply:
Yes, I do agree - I just wanted to check the method renaming changes in under a separate commit.